connections/README.md

```diff
@@ -263,6 +267,49 @@ indicating whether the handler supports the protocol. This allows more flexible
 behavior than exact literal matching, which is the default behavior if no match
 function is provided.
 
+## Stream Behaviour
+
+This section specifies the expected behaviour of a libp2p stream.
+
+### States
+
+A libp2p stream can be in one of four states:
+
+- Open for reading and writing
+- Open for reading only
+- Open for writing only
+- Closed
+
+The default state after opening a new stream is "Open for reading and writing".
+Either party can at any point close the reading or writing side.
+
+The "Closed" state is the product of both halfs (read and write) of the stream being closed.
+This can happen because one party closes both halfs but also as a combination
+of both parties closing their write-half (which automatically closes other's read-half).
+
+Writing to a write-closed stream SHOULD fail with an error clearly signalling the closed state.
+
+Reading from a read-closed stream SHOULD succeed by reading 0 bytes.
+This allows for a robust "read to end"-behaviour.
+Layers on top of the stream MAY fail with an unexpected EOF error in case more data was expected.
+
+### Resets
+
+A stream reset is an abrupt termination of the stream.
+Implementations SHOULD reset a stream to signal a non-graceful close.
+Typically, this is during a resource cleanup like garbage-collection for streams
+which have not been closed properly (see [States](#states) above).
+
+Upon receiving a stream reset, any buffered data in the stream MAY be discarded.
+
+Reading and writing to a reset stream SHOULD fail with an error clearly signalling the reset.
```

Comment:
Cross referencing a past discussion for the curious reader here, namely that QUIC only closes the write side.

https://github.com/libp2p/specs/pull/412#discussion_r983113010

I don't have a strong opinion here. Maybe @marten-seemann does. Ideally we would be consistent across the different multiplexers. At least we should document the difference here.